altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -23,100 +55,204 @@ impl UsartControl {
+    pub fn set_word_length(&mut self, length: WordLength) {
+        self.cr1.set_word_length(length);
     }
 
-    pub fn enable_over8(&self) {
+    /// Makes call to control register one to enable oversampling by 8.
+    pub fn enable_over8(&mut self) {
         self.cr1.set_over8(true);
     }
 
-    pub fn disable_over8(&self) {
+    /// Makes call to control register one to disable oversampling by 8.
+    /// Will default to oversampling by 16.
+    pub fn disable_over8(&mut self) {
         self.cr1.set_over8(false);
     }
 
+    /// Makes call to control register one to check if over8 is enabled.
+    /// Returns true if enabled, false otherwise.
     pub fn get_over8(&self) -> bool {
         self.cr1.get_over8()
     }
 
-    pub fn set_hardware_flow_control(&self, hfc: HardwareFlowControl) {
+    // --------------------------------------------------------------
+    /// This section pertains to calls made to control register two (cr2).
+
+    /// Makes call to control register two to set the number of stop bits.
+    /// Possible StopLength settings are: Half, One, OneAndHalf, Two
+    pub fn set_stop_bits(&mut self, length: StopLength) {
+        self.cr2.set_stop_bits(length);
+    }
+
+    // --------------------------------------------------------------
+    /// This section pertains to calls made to control register three (cr3).
+
+    /// Makes call to control register three to set hardware flow control.
+    /// Possible HardwareFlowControl settings are: None, Rts, Cts, All
+    pub fn set_hardware_flow_control(&mut self, hfc: HardwareFlowControl) {
         self.cr3.set_hardware_flow_control(hfc);
     }
 }
 
 // ------------------------------------
-/// CR1
+/// CR1 - Control Register One
 // ------------------------------------
 
-/// Word length can be 7, 8, or 9 bits.
-#[derive(Copy, Clone)]
-pub enum WordLength {
-    Seven,
-    Eight,
-    Nine,
-}
-
-#[derive(Copy, Clone)]
+/// Defines the possible Mode configurations for the Usart.
+/*  None: Transmit and Receive turned off
+ *  Receive: Usart configured to only receive.
+ *  Transmit: Usart configuration to only transmit.
+ *  All: Transmit and Receive both turned on.
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum Mode {
     None,
     Receive,
     Transmit,
     All,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the possible Parity configurations for the Usart.
+/*  None: No parity configuration set.
+ *  Even: Even parity configuration.
+ *  Odd: Odd parity configuration.
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum Parity {
     None,
     Even,
     Odd,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the possible WordLength configurations for the Usart.
+/*  Seven: Seven bit word length
+ *  Eight: Eight bit word length
+ *  Nine: Nine bit word length
+ */
+#[derive(Copy, Clone, Debug)]
+pub enum WordLength {
+    Seven,
+    Eight,
+    Nine,
+}
+
+/// Defines the base address variable for control register one.
```

Comment:
This isn't a public interface, so we don't strictly need doc comments here, but I would also go into more detail about what it does rather than just storing a value.